tests/components/tedee/test_coordinator.py

```diff
@@ -0,0 +1,37 @@
+"""Tests for tedee lock."""
+import time
+from unittest.mock import MagicMock
+
+from pytedee_async.exception import TedeeDataUpdateException
+import pytest
+
+from homeassistant.components.tedee.const import DOMAIN
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+pytestmark = pytest.mark.usefixtures("init_integration")
+
+
+async def test_coordinator(
+    hass: HomeAssistant,
+    mock_tedee: MagicMock,
+    mock_config_entry: MockConfigEntry,
+    caplog,
+) -> None:
+    """Test coordinator."""
+    mock_tedee.sync.side_effect = TedeeDataUpdateException("")
+    coordinator = hass.data[DOMAIN][mock_config_entry.entry_id]
```

Comment:
We want to avoid testing internals like this. Can we maybe move this test to the lock file and try to use the freezer instead to skip time and make it more "natural"?